src/quote/indexQuoteProvider.ts

```diff
@@ -122,6 +131,48 @@ export class FlashMintQuoteProvider
           tx,
         }
       }
+      case FlashMintContractType.leveragedExtended: {
+        if (!zeroExApiV1) {
+          throw new Error('Contract type requires ZeroExApiV1 to be defined')
+        }
+        const leverageExtendedQuoteProvider =
+          new LeveragedExtendedQuoteProvider(provider, zeroExApiV1)
+        const leveragedExtendedQuote =
+          await leverageExtendedQuoteProvider.getQuote(request)
+        if (!leveragedExtendedQuote) return null
+        const builder = new LeveragedExtendedTransactionBuilder(provider)
+        const txRequest: FlashMintLeveragedExtendedBuildRequest = {
+          isMinting,
+          inputToken: inputToken.address,
+          inputTokenSymbol: inputToken.symbol,
+          outputToken: outputToken.address,
+          outputTokenSymbol: outputToken.symbol,
+          inputTokenAmount: leveragedExtendedQuote.inputTokenAmount,
+          outputTokenAmount: leveragedExtendedQuote.outputTokenAmount,
+          swapDataDebtCollateral: leveragedExtendedQuote.swapDataDebtCollateral,
+          swapDataInputOutputToken: leveragedExtendedQuote.swapDataPaymentToken,
+          swapDataInputTokenForETH:
+            leveragedExtendedQuote.swapDataDebtCollateral, // FIXME: add swapDataInputTokenForETH (returned from quote provider)
+          // FIXME: check if new arguments should be set here in the SDK or rather passed as arguments to they can be set from the app
+          priceEstimateInflator: wei(0.9),
+          maxDust: wei(0.00001),
+        }
+        const tx = await builder.build(txRequest)
+        if (!tx) return null
+        return {
```

Comment:
Make sure to not change anything on the type of this return - as this would change the whole quote providers return type. The app uses this type everywhere atm.